cpp/src/arrow/dataset/scanner.h

```diff
@@ -166,13 +169,29 @@ class ARROW_DS_EXPORT ScanTask {
   std::shared_ptr<Fragment> fragment_;
 };
 
-template <typename T>
-struct Enumerated {
-  T value;
-  int index;
-  bool last;
+/// \brief A trivial ScanTask that yields the RecordBatch of an array.
```

Comment:
FWIW I had moved these helpers to the bottom of the file (outside the `@}`) so that they don't get picked up in the API docs as I'd basically consider them implementation details. Though at that point we may actually want them to be in one of the `_internal` headers.

Reply:
Oh shoot.  After the rebase I had two instances of the exact same method and this explains it.  I'll fix that up.

Reply:
Fixed.

Reply:
Well, for clarity's sake.  I just moved them back to where you had them.  I"ll leave internal headers for a future PR to keep this one at a reasonable size.